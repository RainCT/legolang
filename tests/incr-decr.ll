int i = 5;
print i;    //5
↑i 
print i     //6
int j = ↑i+1
print j     //8

//predecrements
↓i
print i     //6
j = ↓i+1
print j     //6

//post
i↑
print i     //6
j↓
print j     //5
j = i↓ + 1
print j     //7
print ↑i   //6

print "Beginning assignments test!"
int g = 3;
print g     //3
g += 5
print g     //8
g -= 5
print g     //3
g *= 4      
print g     //12
g /= 6
print g     //2
g %= 10
print g     //2
g *= 12
g %= 10
print g     //4

string txt = "Hello"
txt += " world!"
print txt
