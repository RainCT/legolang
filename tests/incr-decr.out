5
6
8
6
6
6
5
7
6
Beginning assignments test!
3
8
3
12
2
2
4
Hello world!
